apiserver/facades/client/machinemanager/machinemanager_test.go

```diff
@@ -168,6 +169,72 @@ func (s *MachineManagerSuite) TestDestroyMachineWithParams(c *gc.C) {
 	})
 }
 
+func (s *MachineManagerSuite) setupUpdateMachineSeries(c *gc.C) {
+	s.st.machines = map[string]*mockMachine{
+		"0": &mockMachine{series: "trusty"},
+		"1": &mockMachine{series: "trusty"},
+	}
+}
+
+func (s *MachineManagerSuite) TestUpdateMachineSeries(c *gc.C) {
+	s.setupUpdateMachineSeries(c)
+	apiV4 := machinemanager.MachineManagerAPIV4{s.api}
+	results, err := apiV4.UpdateMachineSeries(
+		params.UpdateSeriesArgs{
+			Args: []params.UpdateSeriesArg{
+				{
+					Entity: params.Entity{Tag: names.NewMachineTag("0").String()},
+					Series: "xenial",
+				},
+				{
+					Entity: params.Entity{Tag: names.NewMachineTag("1").String()},
+					Series: "trusty",
+				},
+			}},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(len(results.Results), gc.Equals, 2)
+	c.Assert(results.Results[0], jc.DeepEquals, params.ErrorResult{})
```

Comment:
We should add an arg above (and check for the result) for additional cases:
- permission denied
- invalid machine tag
- valid machine tag but unknown machine name
- incompatible series 

It's just a matter of adding args to []params.UpdateSeriesArg and checking for the expected results. I'd change the results check to do a c.Assert(results, jc.DeepEquals, ...)

Reply:
added all but permission denied.  email out on that one.

Did the same for ApplicationSuite.TestApplicationUpdateSeries and added ApplicationSuite.TestUpdateApplicationSeriesIncompatibleSeries too